src/extension.ts

```
@@ -39,34 +41,41 @@ export function activate(context: vscode.ExtensionContext): void {
     registerTableGroupActionHandlers(actionHandler);
 
-    vscode.window.registerTreeDataProvider('azureStorage', azureTreeDataProvider);
-    actionHandler.registerCommand('azureStorage.refresh', (node?: IAzureNode) => azureTreeDataProvider.refresh(node));
+    vscode.window.registerTreeDataProvider('azureStorage', tree);
+    actionHandler.registerCommand('azureStorage.refresh', (node?: IAzureNode) => tree.refresh(node));
     actionHandler.registerCommand('azureStorage.copyUrl', (node?: IAzureNode<IAzureTreeItem & ICopyUrl>) => node.treeItem.copyUrl(node));
     actionHandler.registerCommand('azureStorage.selectSubscriptions', () => commands.executeCommand("azure-account.selectSubscriptions"));
     actionHandler.registerCommand("azureStorage.openInPortal", (node: IAzureNode<IAzureTreeItem>) => {
         node.openInPortal();
-    }
-    );
-    actionHandler.registerCommand("azureStorage.configureStaticWebsite", (node: IAzureNode<IAzureTreeItem>) => {
-        if (node) {
-            let featureQuery = "feature.staticwebsites=true"; // Needed until preview is public
-            let resourceId = `${node.id}/staticWebsite`;
-            node.openInPortal(resourceId, { queryPrefix: featureQuery });
+    });
+    actionHandler.registerCommand("azureStorage.configureStaticWebsite", async (node: IAzureNode<IAzureTreeItem>) => {
+        if (!node) {
+            node = <IAzureNode<StorageAccountNode>>await tree.showNodePicker(StorageAccountNode.contextValue);
+        }
+        if (node.treeItem.contextValue === BlobContainerNode.contextValue) {
+            // Currently the portal only allows configuring at the storage account level, so retrieve the storage account node
+            let storageAccountNode = node.parent && node.parent.parent;
+            console.assert(!!storageAccountNode && storageAccountNode.treeItem.contextValue === StorageAccountNode.contextValue, "Couldn't find storage account node for container");
```

What's the difference between using a console.assert vs. throwing an error?

Also, is there any concern about storageAccountNode.treeItem being undefined and then checking its contextValue?

---

If storageAccountNode.treeItem == undefined, I'd say something is seriously wrong in the tree code and it'll die a horrible death soon enough.  :-)

---

Fair enough 😆 

Do we want to stifle the error with console.assert?  I would think we want to throw an error to at least telemetry.

---

console.assert actually throws an error.

My reasons for using console.assert:
1) It makes it clear that I'm "asserting" something should be true
2) Easier to code/read than an if/else